gnu_dataset/gnats-4.2.0/doc/man/delete-pr.man
.\" Copyright (c) 1999, 2003 Free Software Foundation, Inc.
.\" See section COPYING for conditions for redistribution
.TH delete-pr 8 "August 2003" "GNATS 4.2.0" "GNATS Admininstration Utilities"
.de BP
.sp
.ti -.2i
\(**
..
.SH NAME
delete\-pr \- deletes closed PRs
.SH SYNOPSIS
.hy 0
.na
.TP
.B delete\-pr
[
.B \-V 
|
.B \-\-version
]
|
[
.B \-h
|
.B \-\-help
]
|
.br
[
.BI -d \ name
|
.BI --database= name
]
[
.BR \-c\  | \ \-\-closed\  | \ PR
]
.ad b
.hy 1
.SH DESCRIPTION
Deletes the specified
.BR PR ,
or if the \fB\-\-closed\fR option is given, all of the closed PRs in the
database are deleted.
.P
\fBdelete-pr\fR must be run by the GNATS user (by default \fBgnats\fR).
.P
Note that deleting PRs is generally a \fBvery\fR bad idea.  It flies
in the face of the very idea of using a Problem Reporting system and
should only be considered in special situations.  Such a situation
might for instance arise if a mail loop causes mass duplicate
submissions or if you receive spam to the PR submission e-mail
address.
.SH OPTIONS
.TP
.B \-V\fR,\fB \-\-version
Prints the program version to stdout and exits.
.TP
.B \-h\fR,\fB \-\-help
Prints a short help text to stdout and exits.
.TP
.B \-d\fR,\fB \-\-database
Specifies the database from which the PR(s) are to be deleted.  If no
database is specified, the database named \fIdefault\fR is assumed.
This option overrides the database specified in the \fBGNATSDB\fR
environment variable.
.TP
.B \-\-closed
Requests that all closed PRs in the database be deleted.
.TP
.B PR
Requests that the specified \fBPR\fR be deleted.  The \fBPR\fR must be
in a closed state.
.SH ENVIRONMENT VARIABLES
The \fBGNATSDB\fR environment variable is used to determine which database to
use.  For a local database, it contains the name of the database to access.
.P
For network access via \fIgnatsd\fR, it contains a colon-separated list
of strings that describe the remote database, in the form
.TP
\fIserver\fR:\fIport\fR:\fIdatabasename\fR:\fIusername\fR:\fIpassword\fR
.P
Any of the fields may be omitted, but at least one colon must appear;
otherwise, the value is assumed to be the name of a local database.
.P
If \fBGNATSDB\fR is not set and the \fB--database\fR option is not
supplied, it is assumed that the database is local and that its name
is
\fIdefault\fR.
.SH "SEE ALSO"
.I Keeping Track: Managing Messages With GNATS
(also installed as the GNU Info file
.BR gnats.info )
and
.I Reporting Problems Using send-pr
(also installed as the GNU Info file
.BR send-pr.info ).
.LP
.BR databases (5),
.BR dbconfig (5),
.BR delete-pr (8),
.BR edit-pr (1)
.BR file-pr (8),
.BR gen-index (8),
.BR gnats (7),
.BR gnatsd (8),
.BR mkcat (8),
.BR mkdb (8),
.BR pr-edit (8),
.BR query-pr (1),
.BR queue-pr (8),
.BR send-pr (1).
.SH COPYING
Copyright (c) 1999, 2003, Free Software Foundation, Inc.
.PP
Permission is granted to make and distribute verbatim copies of
this manual provided the copyright notice and this permission notice
are preserved on all copies.
.PP
Permission is granted to copy and distribute modified versions of this
manual under the conditions for verbatim copying, provided that the
entire resulting derived work is distributed under the terms of a
permission notice identical to this one.
.PP
Permission is granted to copy and distribute translations of this
manual into another language, under the above conditions for modified
versions, except that this permission notice may be included in
translations approved by the Free Software Foundation instead of in
the original English.


gnu_dataset/gnats-4.2.0/gnats/delete-pr.sh
#! /bin/sh
# Program to delete problem reports for GNATS.
# Copyright (C) 2007  Free Software Foundation, Inc.
# 
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with GNU GNATS; see the file COPYING. If not, see
# <http://www.gnu.org/licenses/>.
#
# Contributed by Kevin Hopkins (K.Hopkins@cs.nott.ac.uk).

PATH="@pkglibexecdir@:@sbindir@:@bindir@:/bin:/usr/bin:${PATH}"
export PATH

GNATS_USER="@GNATS_USER@"

locked=
closed=
closed_state=

usage="Usage: $0 [-fhV] [--version] [--help] [ -c | --closed | PR ]"
version="@VERSION@"

# parse command line.
# for the non-flag argument, assume it's correct.  if it's a full
# id number, use that; if not, find the full id.
# only continue if $full_id is an actual file.

if [ $# -eq 0 ]; then
  echo "$usage" ; exit 1
fi
case "$1" in
  -V|-v|--version|--ve*)
    echo "$version"; exit 0
    ;;
  -h|--help*)
    echo "$usage"; exit 0
    ;;
  -d|--database)
    if [ "$#" -le 1 ] ; then echo "$usage"; exit 1; fi
    shift ; GNATS_DB="--database=$1" ;;
  -d=*|--database=*) GNATS_DB="$1" ;;
  -c|--closed)
    if [ "$#" -eq "1" ] ; then
      closed=t
    else
      echo "$usage"; exit 1
    fi
    ;;
  -*)
    echo "$usage"; exit 1
    ;;
  *)
    pr=$1
    ;;
esac

if [ "$closed" = "t" ]
then
    prs=`query-pr $GNATS_DB --format Number --state closed`
else
    if [ "x$pr" = "x" ]
    then
	echo "Must specify either a PR to delete or the --closed option.";
	exit 1
    fi
    prs=$pr
fi

if [ "`echo -n`" = "-n" ]
then	echon () { echo $* \\c ; }
else	echon () { echo -n $* ; }
fi

if [ -z "$HOSTNAME" ]; then
  if [ -f /bin/hostname ] ; then HOSTNAME=`/bin/hostname`
  elif [ -f /usr/bin/hostname ] ; then HOSTNAME=`/usr/bin/hostname`
  # Solaris et al.
  elif [ -f /usr/ucb/hostname ] ; then HOSTNAME=`/usr/ucb/hostname`
  # Irix
  elif [ -f /usr/bsd/hostname ] ; then HOSTNAME=`/usr/bsd/hostname`
  fi
fi

# find a user name
if [ "$USER" != "" ]; then
  me=$USER
else
  if [ "$LOGNAME" != "" ]; then
    me=$LOGNAME
  else
    echo "edit-pr: no user name found---set LOGNAME." ; exit 1
  fi
fi

if [ "x$me" != "x$GNATS_USER" ]
then
    echo "To delete a PR, you must be $GNATS_USER";
    exit 1
fi

if [ -n "$HOSTNAME" ]; then
  full_me="$me@$HOSTNAME"
else
  full_me="$me"
fi

# start of loop
for prnum in $prs
do
    # XXX ??? !!! Should use State[type] here instead, but then we'd have
    # to change the check for a valid PR number, because non-closed states
    # are usually returned as the empty string for a type.
    state=`query-pr $GNATS_DB --format State $prnum`

    case $state in
	"")
	    echo "No such PR as $prnum" 1>&2
	    continue
	    ;;
	[Cc]losed)	closed_state=t ;;
	*)		closed_state="" ;;
    esac

    if [ "$closed_state" = "t" ]
    then
	echon "Do you want to delete problem $prnum? [n]"
	read answer junk < /dev/tty
	case $answer in
	    [yY]*)
		;;
	    [qQ]*)
		exit 0
		;;
	    *)
		continue;
	        ;;
	esac
    else
	echo "To delete $prnum, it must be closed.";
	exit 1
    fi

    pr-edit $GNATS_DB --email-addr "$full_me" --delete-pr $prnum
    if [ $? != 0 ]
    then
	continue;
    fi
# end of loop
done

exit 0
